### PR TITLE
Adds section on metrics registration

### PR DESCRIPTION
**Pull request by jbee**
Signed-off-by: Jan Bernitt <jaanbernitt@gmail.com>

### CODE REVIEW

**Review by Emily-Jiang — APPROVED** [review]
LGTM!

File: spec/src/main/asciidoc/metrics.asciidoc [hunk 1]
```diff
@@ -34,6 +35,11 @@ is non-portable and may vary between implementations. For portable behavior, mon
 
 Metrics added by this specification will appear in the `base` MicroProfile Metrics scope.
 
+=== Registration
+
+All metrics added by this specification are registered either on startup or on first call of the annotated method. 
+Policies that have been disabled though configuration do not cause registration of the corresponding metrics.
```
Comment (Emily-Jiang) [inline spec/src/main/asciidoc/metrics.asciidoc:41]: though -> through

Comment (jbee) [inline spec/src/main/asciidoc/metrics.asciidoc:41]:
```suggestion
Policies that have been disabled through configuration do not cause registration of the corresponding metrics.
```

**Review by Azquelt — APPROVED** [review]

**Review by Ladicek — APPROVED** [review]

**Comment by jbee** [comment]
Came to think about permutations of possible tags for some of the metrics. Eager approach really does not make much sense there.

**Review by Azquelt — APPROVED** [review]

**Comment by Azquelt** [comment]
> Came to think about permutations of possible tags for some of the metrics

We discussed this and ended up deciding that it did make sense to eagerly create metrics for all tag combinations which could apply to the method in question, hence the update in the second commit.

**Pull request merged**